<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>KeyDesignUI</class>
 <widget class="QMainWindow" name="KeyDesignUI">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>588</width>
    <height>414</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>键UI设计</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QVBoxLayout" name="verticalLayout_5">
    <item>
     <widget class="QSplitter" name="splitter_2">
      <property name="orientation">
       <enum>Qt::Orientation::Horizontal</enum>
      </property>
      <widget class="QScrollArea" name="scrollArea">
       <property name="widgetResizable">
        <bool>true</bool>
       </property>
       <widget class="QWidget" name="scrollAreaWidgetContents">
        <property name="geometry">
         <rect>
          <x>0</x>
          <y>0</y>
          <width>288</width>
          <height>341</height>
         </rect>
        </property>
        <layout class="QVBoxLayout" name="verticalLayout_4">
         <item>
          <widget class="QGroupBox" name="groupBox_4">
           <property name="title">
            <string>设计参数(一)</string>
           </property>
           <layout class="QGridLayout" name="gridLayout_2" columnstretch="1,1,1,1">
            <item row="1" column="3">
             <widget class="QDoubleSpinBox" name="spin_L">
              <property name="decimals">
               <number>0</number>
              </property>
              <property name="maximum">
               <double>5000.000000000000000</double>
              </property>
              <property name="value">
               <double>90.000000000000000</double>
              </property>
             </widget>
            </item>
            <item row="0" column="3">
             <widget class="QDoubleSpinBox" name="spin_d">
              <property name="decimals">
               <number>0</number>
              </property>
              <property name="maximum">
               <double>5000.000000000000000</double>
              </property>
              <property name="value">
               <double>70.000000000000000</double>
              </property>
             </widget>
            </item>
            <item row="1" column="0">
             <widget class="QPushButton" name="btn_h">
              <property name="text">
               <string>键高(h)</string>
              </property>
             </widget>
            </item>
            <item row="1" column="2">
             <widget class="QPushButton" name="btn_L">
              <property name="text">
               <string>键长(L)</string>
              </property>
             </widget>
            </item>
            <item row="0" column="1">
             <widget class="QDoubleSpinBox" name="spin_b">
              <property name="decimals">
               <number>0</number>
              </property>
              <property name="maximum">
               <double>5000.000000000000000</double>
              </property>
              <property name="value">
               <double>20.000000000000000</double>
              </property>
             </widget>
            </item>
            <item row="0" column="0">
             <widget class="QPushButton" name="btn_b">
              <property name="text">
               <string>键宽(b)</string>
              </property>
             </widget>
            </item>
            <item row="1" column="1">
             <widget class="QDoubleSpinBox" name="spin_h">
              <property name="decimals">
               <number>0</number>
              </property>
              <property name="value">
               <double>12.000000000000000</double>
              </property>
             </widget>
            </item>
            <item row="2" column="1">
             <widget class="QDoubleSpinBox" name="spinObs">
              <property name="decimals">
               <number>0</number>
              </property>
              <property name="maximum">
               <double>500.000000000000000</double>
              </property>
              <property name="value">
               <double>110.000000000000000</double>
              </property>
             </widget>
            </item>
            <item row="2" column="0">
             <widget class="QPushButton" name="btnObs">
              <property name="text">
               <string>许用挤压应力(Mpa)</string>
              </property>
             </widget>
            </item>
            <item row="2" column="2">
             <widget class="QPushButton" name="btnLoad">
              <property name="text">
               <string>载荷性质</string>
              </property>
             </widget>
            </item>
            <item row="2" column="3">
             <widget class="QComboBox" name="comboBox">
              <property name="currentIndex">
               <number>1</number>
              </property>
              <item>
               <property name="text">
                <string>静载荷</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string>轻微冲击</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string>冲击</string>
               </property>
              </item>
             </widget>
            </item>
            <item row="0" column="2">
             <widget class="QPushButton" name="btn_d">
              <property name="text">
               <string>轴的直径(d)</string>
              </property>
             </widget>
            </item>
           </layout>
          </widget>
         </item>
         <item>
          <widget class="QLabel" name="label_4">
           <property name="font">
            <font>
             <pointsize>12</pointsize>
             <bold>true</bold>
            </font>
           </property>
           <property name="text">
            <string>点击按钮，即可进行查表操作，键长要小于轴段长度</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QGroupBox" name="groupBox_3">
           <property name="title">
            <string>设计参数(二)</string>
           </property>
           <layout class="QGridLayout" name="gridLayout" columnstretch="1,1,1,1">
            <item row="0" column="0">
             <widget class="QLabel" name="label">
              <property name="text">
               <string>轴转速(r/min)</string>
              </property>
             </widget>
            </item>
            <item row="0" column="1">
             <widget class="QDoubleSpinBox" name="spin_n">
              <property name="maximum">
               <double>5000.000000000000000</double>
              </property>
              <property name="value">
               <double>970.000000000000000</double>
              </property>
             </widget>
            </item>
            <item row="0" column="2" colspan="2">
             <widget class="QPushButton" name="btnCalcT">
              <property name="text">
               <string>计算转矩(直接输入转矩也行)</string>
              </property>
             </widget>
            </item>
            <item row="1" column="0">
             <widget class="QLabel" name="label_2">
              <property name="text">
               <string>传递的功率(KW)</string>
              </property>
             </widget>
            </item>
            <item row="1" column="1">
             <widget class="QDoubleSpinBox" name="spin_P">
              <property name="value">
               <double>11.000000000000000</double>
              </property>
             </widget>
            </item>
            <item row="1" column="2">
             <widget class="QLabel" name="label_3">
              <property name="text">
               <string>传递的转矩(N*m)</string>
              </property>
             </widget>
            </item>
            <item row="1" column="3">
             <widget class="QDoubleSpinBox" name="spin_T">
              <property name="decimals">
               <number>1</number>
              </property>
              <property name="maximum">
               <double>5000.000000000000000</double>
              </property>
              <property name="value">
               <double>2200.000000000000000</double>
              </property>
             </widget>
            </item>
           </layout>
          </widget>
         </item>
         <item>
          <widget class="QLabel" name="label_5">
           <property name="font">
            <font>
             <pointsize>12</pointsize>
             <bold>true</bold>
            </font>
           </property>
           <property name="text">
            <string>支持平键A型设计，其他类型暂不支持</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QGroupBox" name="groupBox_5">
           <property name="title">
            <string>拓展部分</string>
           </property>
           <layout class="QGridLayout" name="gridLayout_3">
            <item row="0" column="0">
             <widget class="QPushButton" name="btn_FlowerO">
              <property name="text">
               <string>花键的需用应力</string>
              </property>
             </widget>
            </item>
            <item row="0" column="1">
             <widget class="QPushButton" name="btn_calFlower">
              <property name="text">
               <string>花键的校核流程</string>
              </property>
             </widget>
            </item>
            <item row="1" column="0">
             <widget class="QPushButton" name="btn_FatPara">
              <property name="text">
               <string>胀套的额定载荷系数</string>
              </property>
             </widget>
            </item>
            <item row="1" column="1">
             <widget class="QPushButton" name="btn_calFat">
              <property name="text">
               <string>胀套的校核流程</string>
              </property>
             </widget>
            </item>
           </layout>
          </widget>
         </item>
        </layout>
       </widget>
      </widget>
      <widget class="QSplitter" name="splitter">
       <property name="orientation">
        <enum>Qt::Orientation::Vertical</enum>
       </property>
       <widget class="QGroupBox" name="groupBox">
        <property name="title">
         <string>设计图表</string>
        </property>
        <layout class="QVBoxLayout" name="verticalLayout">
         <item>
          <widget class="QScrollArea" name="scrollArea_2">
           <property name="widgetResizable">
            <bool>true</bool>
           </property>
           <widget class="QWidget" name="scrollAreaWidgetContents_2">
            <property name="geometry">
             <rect>
              <x>0</x>
              <y>0</y>
              <width>811</width>
              <height>315</height>
             </rect>
            </property>
            <layout class="QVBoxLayout" name="verticalLayout_3">
             <item>
              <widget class="QLabel" name="labPic">
               <property name="text">
                <string/>
               </property>
               <property name="pixmap">
                <pixmap resource="res.qrc">:/images/images/Key_Size.png</pixmap>
               </property>
              </widget>
             </item>
            </layout>
           </widget>
          </widget>
         </item>
        </layout>
       </widget>
       <widget class="QGroupBox" name="groupBox_2">
        <property name="title">
         <string>输出文本框</string>
        </property>
        <layout class="QVBoxLayout" name="verticalLayout_2">
         <item>
          <widget class="QPlainTextEdit" name="textEdit"/>
         </item>
        </layout>
       </widget>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QToolBar" name="toolBar">
   <property name="windowTitle">
    <string>toolBar</string>
   </property>
   <property name="toolButtonStyle">
    <enum>Qt::ToolButtonStyle::ToolButtonTextUnderIcon</enum>
   </property>
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
   <addaction name="actStart"/>
   <addaction name="separator"/>
   <addaction name="actShow"/>
   <addaction name="actClear"/>
   <addaction name="separator"/>
   <addaction name="actExit"/>
  </widget>
  <action name="actStart">
   <property name="icon">
    <iconset resource="res.qrc">
     <normaloff>:/images/images/wizard24.png</normaloff>:/images/images/wizard24.png</iconset>
   </property>
   <property name="text">
    <string>开始设计</string>
   </property>
   <property name="toolTip">
    <string>开始设计</string>
   </property>
   <property name="menuRole">
    <enum>QAction::MenuRole::NoRole</enum>
   </property>
  </action>
  <action name="actShow">
   <property name="icon">
    <iconset resource="res.qrc">
     <normaloff>:/images/images/704.bmp</normaloff>:/images/images/704.bmp</iconset>
   </property>
   <property name="text">
    <string>显示设计</string>
   </property>
   <property name="toolTip">
    <string>显示设计</string>
   </property>
   <property name="menuRole">
    <enum>QAction::MenuRole::NoRole</enum>
   </property>
  </action>
  <action name="actClear">
   <property name="icon">
    <iconset resource="res.qrc">
     <normaloff>:/images/images/103.bmp</normaloff>:/images/images/103.bmp</iconset>
   </property>
   <property name="text">
    <string>清除输出</string>
   </property>
   <property name="toolTip">
    <string>清空文本框</string>
   </property>
   <property name="menuRole">
    <enum>QAction::MenuRole::NoRole</enum>
   </property>
  </action>
  <action name="actExit">
   <property name="icon">
    <iconset resource="res.qrc">
     <normaloff>:/images/images/exit.bmp</normaloff>:/images/images/exit.bmp</iconset>
   </property>
   <property name="text">
    <string>退出</string>
   </property>
   <property name="toolTip">
    <string>退出程序</string>
   </property>
   <property name="menuRole">
    <enum>QAction::MenuRole::NoRole</enum>
   </property>
  </action>
 </widget>
 <resources>
  <include location="res.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>actExit</sender>
   <signal>triggered()</signal>
   <receiver>KeyDesignUI</receiver>
   <slot>close()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>399</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
